feat(inference): adding delete button

axel7083 opened the pull request:
### What does this PR do?

This PR is adding the logic to delete an InferenceServer. 

> To enhance the UI we will need https://github.com/projectatomic/ai-studio/issues/547 to store the _transition_ states

### Screenshot / video of UI

https://github.com/projectatomic/ai-studio/assets/42176370/947f142d-6598-4d0d-a216-1b85b8b12421

### What issues does this PR fix or reference?

Fixes https://github.com/projectatomic/ai-studio/issues/546

### How to test this PR?

- [x] unit tests has been provided

jeffmaury reviewed (changes requested):
Missing confirmation dialog

axel7083 commented:
> Missing confirmation dialog

@jeffmaury good remarks I rebased and added a confirmation dialog as visible in the following video

https://github.com/projectatomic/ai-studio/assets/42176370/22e122a7-e641-4822-96c0-e8d64aa293df

jeffmaury commented on packages/backend/src/managers/inference/inferenceManager.ts:
```diff
@@ -306,14 +306,40 @@ export class InferenceManager extends Publisher<InferenceServer[]> implements Di
+   * @param containerId
    */
-  private removeInferenceServer(containerId: string): void {
+  private removeInferenceServerRef(containerId: string): void {
```
Rename as removeInferenceServerReference

axel7083 replied:
✅  done

jeffmaury reviewed (approved):
LGTM comment wise
